bot/handlers/role_selection.go

```diff
@@ -10,8 +10,20 @@ import (
 	"go.uber.org/zap"
 )
 
+// DiscordSession is an interface for `discordgo.Session` to allow mocking.
+type DiscordSession interface {
```

Comment:
@jules I don't know that you need a new interface here. You may be able to mock discordgo.Session in the test file by initializing an empty struct. Is there a reason for this particular approach?